sdks/java/core/src/main/java/org/apache/beam/sdk/state/Timer.java

```diff
@@ -81,6 +81,9 @@
    */
```

Comment:
We should add documentation to Timer saying that set and/or clear calls may only become visible after this bundle completes and may not be applied immediately allowing for existing timers which have become eligible to still fire.

Reply:
But that is incorrect behavior (as has been agreed on the dev list many times). set/clear calls should take effect immediately. The fact that this is sometimes not true has caused user bugs before because calls to modify state take effect immediately, leading to users seeing inconsistencies between state & timers.

I have a followon PR to fix this bug, so set/clear calls will be visible immediately.

Reply:
I see, if that is the case then great, we should add the tests as I suggested below.

Reply:
Will add this test to the followon PR, as it will fail until that is in

Reply:
sg